hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -413,16 +407,15 @@ fanoutTx utxo (headInput, headOutput, ScriptDatumForTxIn -> headDatumBefore) (sl
     BuildTxWith $ ScriptWitness scriptWitnessCtx $ mkScriptWitness headScript headDatumBefore headRedeemer
 
   headScript =
-    fromPlutusScript @PlutusScriptV1 Head.validatorScript
-
+    fromPlutusScript @PlutusScriptV2 Head.validatorScript
   headRedeemer =
     toScriptData (Head.Fanout $ fromIntegral $ length utxo)
 
   headTokens =
     headTokensFromValue headTokenScript (txOutValue headOutput)
 
   fanoutOutputs =
-    map toTxContext $ toList utxo
+    map (toTxContext . snd) . sortOn fst $ UTxO.pairs utxo
```

Comment:
This change is in principle semantically equivalent, because `toList` returns elements ordered by key already. So perhaps we could make the naming clearer or add a comment to clarify this.